### PR TITLE
Project structure change (Closes #4)

### LINKED ISSUE #4: Revamp Project Structure
We will be using the below guide and following the structure with some modifications.

Reference:
- https://www.freecodecamp.org/news/structuring-a-flask-restplus-web-service-for-production-builds-c2ec676de563/
- https://medium.com/ki-labs-engineering/designing-well-structured-rest-apis-with-flask-restplus-part-1-7e96f2da8850
- http://michal.karzynski.pl/blog/2016/06/19/building-beautiful-restful-apis-using-flask-swagger-ui-flask-restplus/

**Deadline:** Sunday 6PM

### PR DESCRIPTION
**Pull request by thescriptninja**
1. Changed the project structure as mentioned in the [tutorial](https://www.freecodecamp.org/news/structuring-a-flask-restplus-web-service-for-production-builds-c2ec676de563/).
Used this [repo](https://github.com/cosmic-byte/flask-restplus-boilerplate) as reference as mentioned in the article.
2. Removed `requirements.txt` file

### CODE REVIEW

**Comment by thescriptninja** [comment]
I think this PR ready. :v:

**Comment by mukul-mehta** [comment]
Hi, @thealphadollar and @thescriptninja, please go through #12. I've tried to make the changes suggested.

**Comment by thescriptninja** [comment]
### Makefile added. 
A bit confused about running pipenv, pip3 and python3 in the make command. @thealphadollar can you review this?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+clean:
+	find . -type f -name '*.pyc' -delete
+	find . -type f -name '*.log' -delete
+
+system-packages: 
+	sudo apt install python3-pip -y
+	pip3 install pipenv
+
+python-packages:
+	pipenv install
```
Comment (thealphadollar) [inline Makefile]: `pipenv install --dev` since the makefile will be used by us.

File: requirements.txt [hunk 2]
```diff
@@ -1,15 +0,0 @@
-aniso8601==7.0.0
```
Comment (thealphadollar) [inline requirements.txt]: Do not remove this for now.

File: README.md [hunk 3]
```diff
@@ -39,10 +39,3 @@ This will update Pipfile and Pipfile.lock
 
 To upgrade a package, use ` pipenv update <pkg-name> `.
 
-# Adding new dependencies to requirements.txt
```
Comment (thealphadollar) [inline README.md]: Do not remove this.

File: README.md [hunk 4]
```diff
@@ -39,10 +39,3 @@ This will update Pipfile and Pipfile.lock
...
-
-Whenever a new dependency is used, be sure to run 
-````
-pip freeze > requirements.txt
-````
-Mention dependency change in the commit message.
```
Comment (thealphadollar) [inline README.md]: Please add instructions on how to perform various actions using the makefile.

**Comment by thescriptninja** [comment]
Changes done as requested :+1:

File: README.md [hunk 5]
```diff
@@ -5,7 +5,7 @@ This is the common backend for Tech-Ambit web and android application.
 # Directory structure
 ```` 
     .
-    |──rest_apis/
+    ├──rest_apis/
```
Comment (thealphadollar) [inline README.md]: We've updated the structure, right? That should be reflected here as well.

File: README.md [hunk 6]
```diff
@@ -25,24 +25,47 @@ This is the common backend for Tech-Ambit web and android application.
 
 This can improved though.
 
-Collect all similar endpoints in 1 file in `endpoints` folder. Use different files for logically dissimilar endpoints.
+Collect all similar endpoints in 1 file in `app/controller` folder. Use different files for logically dissimilar endpoints.
```
Comment (thealphadollar) [inline README.md]: "1 file" -> "the folder"

File: app/main/__init__.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+from .config import config_by_name
+
```
[the remark this answers is not in the document]
Comment (thealphadollar) [reply app/main/__init__.py]: @thescriptninja This?

[the remark this answers is not in the document]
Comment (thescriptninja) [reply app/main/__init__.py]: @mukul-mehta has it in his PR. Should I still add it?

[the remark this answers is not in the document]
Comment (thealphadollar) [reply app/main/__init__.py]: No.

File: README.md (hunk 5 above)
Comment (thescriptninja) [reply README.md]: Yeah I'll do this

**Comment by mukul-mehta** [comment]
Ok, @thealphadollar . I'll add restplus to my code and create a new PR.

**Pull request merged**